### PR TITLE
[FLINK-29246] Adding skeleton project files and checkstyle configuration

### PR DESCRIPTION
**Pull request by dannycranmer**
## What is the purpose of the change

Adding the baseline project structure, Maven pom files and checkstyle configuration

## Brief change log

* Added parent, connector and SQL connector `pom.xml` files
* Adding checkstyle config (from Flink) and empty suppressions file

## Verifying this change

- `mvn clean package` is successfull

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? n/a

### CODE REVIEW

**Review by z3d1k — APPROVED** [review]
LGTM

File: tools/maven/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+		<property name="severity" value="error"/>
+	</module>
+
+	<module name="RegexpSingleline">
+		<property name="format" value="Throwables.propagate\("/>
+		<property name="message" value="Throwables.propagate is deprecated"/>
+		<property name="severity" value="error"/>
+	</module>
+
+	<!-- Prevent *Tests.java as tools may not pick them up -->
```
Comment (mrapcooper) [inline tools/maven/checkstyle.xml:58]: There is a lot of commented out config in this new file. Why?

Comment (dannycranmer) [reply tools/maven/checkstyle.xml:58]: The checkstyle config is a direct copy from Flink. I would rather copy as is so it is easier to keep up to date. Ideally I would just reference it, will see if I can set that up

Comment (dannycranmer) [reply tools/maven/checkstyle.xml:58]: I tried referencing but could not get it to work, will cycle back to this later.

**Comment by dannycranmer** [comment]
Github actions tested in separate dummy PR https://github.com/apache/flink-connector-dynamodb/pull/3/checks

**Pull request merged**